### DISCUSSION TITLE
Will templated files get auto-updated in the future?

### DISCUSSION THREAD

**Discussion by kyounger** [zimfw/zimfw#455] [Q&A] [answered]
Hello. The zimrc and zshrc files that are created during the installation process -- will those ever get automatically updated by some future process? I noticed they have boundary makers in them, like this:

`# Start configuration added by Zim install {{"{{{"}}`

This sort of makes me expect that it will get updated programmatically?

For the zshrc template... is the only truly important section the module init?

My second question then would be: What custom config should go before the module init section and what goes after it? Are there things that _have_ to go in either section? E.g. where should `setopt`s go? I see I need certain zstyles for module config before the module init. What about a set of custom shell aliases? Should I source those after the module init?

**Comment by ericbn** [comment, marked as answer]
Hi @kyounger. Very good questions!

> This sort of makes me expect that it will get updated programmatically?

There are no current plans to automatically update any section added to the dot files by the installation script.

> For the zshrc template... is the only truly important section the module init?

Yes. I'm trying to update the documentation to make it clear what all the sections in the `~/.zshrc` do. Comments in https://github.com/zimfw/zimfw/pull/453 are more than welcome!

> What custom config should go before the module init section and what goes after it?

This really depends on each specific custom config case. `zstyle` definitions tend to be required before the modules are initialized. Also, `fpath` must be set to its final value before the `completion` module is initialized, at least for the paths in `fpath` that could contain completion definitions. Everything that depends on the module being already initialized must, of course, be done after that. Lastly, everything that you want to override must, of course, be defined after modules are initialized.

Maybe I'm missing some other obvious examples. Anyone?

**Comment by pdufour** [comment]
Hmm @ericbn I would really suggest maybe only adding what's necessary and then maybe a link to a example of more advanced things you can do, I think zshrc's can get very long when you start having a lot of things going on and keeping the zimfw part short would be a big plus for it's usage IMO.
